src/librustdoc/lib.rs

```diff
@@ -94,6 +94,7 @@ pub fn main() {
         32_000_000 // 32MB on other platforms
     };
     rustc_driver::set_sigpipe_handler();
+    rustc_driver::install_ice_hook();
```

Comment:
This will print a query stack on panic. Previously it would only print a backtrace, which made it really hard to figure out what queries were being executed.